apps/builder/app/canvas/features/webstudio-component/get-browser-style.ts

```diff
@@ -69,3 +93,99 @@ export const getBrowserStyle = (element?: Element): Style => {
   }
   return browserStyle;
 };
+
+// Init with some defaults to avoid undefined
+const defaultUnitSizes: UnitSizes = {
+  ch: 8,
+  vw: 3.2,
+  vh: 4.8,
+  em: 16,
+  rem: 16,
+  px: 1,
+};
+
+export const calculateUnitSizes = (
+  instanceSelector: InstanceSelector | undefined
+): UnitSizes => {
+  if (instanceSelector === undefined) {
+    return defaultUnitSizes;
+  }
+
+  const elements = getAllElementsByInstanceSelector(instanceSelector);
+
+  if (elements.length === 0) {
+    return defaultUnitSizes;
+  }
+
+  const element = elements[0];
+
+  if (element === undefined) {
+    return defaultUnitSizes;
+  }
+
+  // Based on this https://stackoverflow.com/questions/1248081/how-to-get-the-browser-viewport-dimensions/8876069#8876069
+  // this is crossbrowser way to get viewport sizes vw vh in px
+  const vw =
+    Math.max(document.documentElement.clientWidth, window.innerWidth) / 100;
```

Comment:
this would probably be more accurate if it used some element as a probe, because on mobile, these vh,vw and the newer units that react to keyboard etc is hard to calculate in js

Reply:
not this PR issue, lets skip

Reply:
I would rather try to put a probe on canvas for any of this detection logic and let css/dom do it